tests/test_handlers.py

```diff
@@ -750,5 +771,427 @@ def setUp(self) -> None:
+
+            # Check auto generated kv pairs
+            self.assertAlmostEqual(
+                auto_generated_kv_pairs[ZONED_TIMESTAMP_KEY][ZONED_TIMESTAMP_UTC_EPOCH_MS_KEY],
+                expected.utc_epoch_ms,
+                delta=ASSERT_TIMESTAMP_DELTA_MS,
+            )
+            self.assertEqual(
+                auto_generated_kv_pairs[ZONED_TIMESTAMP_KEY][ZONED_TIMESTAMP_TZ_KEY], expected.tz
+            )
+
+            self.assertEqual(auto_generated_kv_pairs[LEVEL_KEY][LEVEL_NO_KEY], expected.level_no)
+            self.assertEqual(
+                auto_generated_kv_pairs[LEVEL_KEY][LEVEL_NAME_KEY], expected.level_name
+            )
+
+            # Check the path by converting the path string to `Path` object
+            self.assertEqual(
+                Path(auto_generated_kv_pairs[SOURCE_CONTEXT_KEY][SOURCE_CONTEXT_PATH_KEY]),
+                expected.path,
+            )
+            if expected.line is not None:
+                self.assertEqual(
+                    auto_generated_kv_pairs[SOURCE_CONTEXT_KEY][SOURCE_CONTEXT_LINE_KEY],
+                    expected.line,
+                )
+
+    def _read_clp(self) -> List[Dict[str, Any]]:
+        result: List[Dict[str, Any]] = []
+        with open(str(self._clp_log_path), "rb") as ir_stream:
+            deserializer: Deserializer = Deserializer(ir_stream)
+            while True:
+                deserialized_log_event: Optional[KeyValuePairLogEvent] = (
+                    deserializer.deserialize_log_event()
+                )
+                if deserialized_log_event is None:
+                    break
+                result.append(deserialized_log_event.to_dict())
+        return result
+
+
+class TestClpKeyValuePairHandlerBase(TestClpKeyValuePairLoggingBase):
+    def test_basic(self) -> None:
+        static_message: Dict[str, str] = {"static_message": "This is a static message"}
+        primitive_dict: Dict[str, Any] = {
+            "str": "str",
+            "int": 1,
+            "float": 1.0,
+            "bool": True,
+            "null": None,
+        }
+        primitive_array: List[Any] = ["str", 1, 1.0, True, None]
+
+        self._log(logging.DEBUG, {"message": f"user_id={self.id()}", "static": static_message})
+        self._log(logging.INFO, {"dict": primitive_dict})
+        self._log(logging.WARNING, {"array": primitive_array})
+        self._log(logging.ERROR, {"array": primitive_array})
+
+        dict_with_array: Dict[str, Any] = copy.deepcopy(primitive_dict)
+        dict_with_array["array"] = primitive_array
+        array_with_dict: List[Any] = copy.deepcopy(primitive_array)
+        array_with_dict.append(primitive_dict)
+        self._log(
+            logging.CRITICAL,
+            {"dict_with_array": dict_with_array, "array_with_dict": array_with_dict},
+        )
+
+        self._close_and_compare()
+
+    def test_empty(self) -> None:
```

Comment:
Can we also test creating an entirely empty file? I.e., one without a log event.